### PR TITLE
Feature/cvsb 11833

### PR DESCRIPTION
**Pull request by KHodderDVSA**
unification of any changes from 10751, 10752 and 10754

### CODE REVIEW

File: src/models/TechRecordsDAO.ts [hunk 1]
```diff
@@ -28,34 +27,44 @@ class TechRecordsDAO {
...
       TableName: this.tableName,
+      IndexName: "",
       KeyConditionExpression: "",
       ExpressionAttributeNames: {},
       ExpressionAttributeValues: {}
     };
 
-    if (isVinSearch(searchTerm, searchCriteria)) { // Query for a full VIN
-      // The partial VIN is a primary index, and is always required
+    if (isSysNumSearch(searchCriteria)) { // Query for a full VIN
```
[the remark this answers is not in the document]
Comment (KHodderDVSA) [reply src/models/TechRecordsDAO.ts]: fixed

File: src/functions/getTechRecords.ts [hunk 2]
```diff
@@ -23,11 +25,24 @@ const getTechRecords = (event: any) => {
...
+  // TODO Not currently used. Probably should be. isValidSearchCriteria() just returns true to bypass at  the moment.
+  if(!isValidSearchCriteria(searchCriteria)) {
+    return Promise.resolve(new HTTPResponse(400, "The search criteria specified is not valid."));
+  }
+
   return techRecordsService.getTechRecordsList(searchIdentifier, status, searchCriteria)
-    .then((data: ITechRecord[]) => {
+    .then((data: ITechRecordWrapper | ITechRecordWrapper[]) => {
+
+      if(!(data instanceof Array)) {
```
[the remark this answers is not in the document]
Comment (KHodderDVSA) [reply src/functions/getTechRecords.ts]: fixed

File: src/functions/getTechRecords.ts [hunk 3]
```diff
@@ -23,11 +25,24 @@ const getTechRecords = (event: any) => {
...
   }
 
+  // TODO Not currently used. Probably should be. isValidSearchCriteria() just returns true to bypass at  the moment.
+  if(!isValidSearchCriteria(searchCriteria)) {
+    return Promise.resolve(new HTTPResponse(400, "The search criteria specified is not valid."));
+  }
+
   return techRecordsService.getTechRecordsList(searchIdentifier, status, searchCriteria)
-    .then((data: ITechRecord[]) => {
+    .then((data: ITechRecordWrapper | ITechRecordWrapper[]) => {
```
[the remark this answers is not in the document]
Comment (KHodderDVSA) [reply src/functions/getTechRecords.ts]: fixed

File: src/utils/PayloadValidation.ts [hunk 4]
```diff
@@ -38,3 +39,9 @@ export const populateFields = (techRecord: ITechRecord) => {
   techRecord.bodyType.code = populateBodyTypeCode(techRecord.bodyType.description);
   techRecord.vehicleClass.code = populateVehicleClassCode(techRecord.vehicleClass.description);
 };
+
+export const isValidSearchCriteria = (specifiedCriteria: string): boolean => {
+  const vals: string[] = Object.values(SEARCHCRITERIA);
+  // return vals.includes(specifiedCriteria); //TODO reinstate for proper input validation
```
Comment (agoncearenco) [inline src/utils/PayloadValidation.ts]: Is this TODO still valid?

Comment (KHodderDVSA) [reply src/utils/PayloadValidation.ts]: Yes. This really should be switched on, but there's no scope for testing it in current stories, so opted to leave it off for now.

**Pull request merged**